config/tools/config_generator.cpp

```diff
@@ -0,0 +1,897 @@
+        -> std::pair<std::string, std::string> {
+        cbdc::privkey_t seckey;
+        for(auto&& b : seckey) {
+            b = rand();
+        }
+        cbdc::pubkey_t ret = cbdc::pubkey_from_privkey(seckey, m_secp.get());
+        std::string seckey_str = cbdc::to_string(seckey);
+        std::string pubkey_str = cbdc::to_string(ret);
+        return std::make_pair(seckey_str, pubkey_str);
+    }
+
+    [[nodiscard]] auto config_generator::create_random_key_pair()
+        -> std::pair<std::string, std::string> {
+        std::uniform_int_distribution<unsigned char> keygen;
+
+        cbdc::privkey_t seckey;
+        for(auto&& b : seckey) {
+            b = keygen(*m_random_source);
+        }
+        cbdc::pubkey_t ret = cbdc::pubkey_from_privkey(seckey, m_secp.get());
+        std::string seckey_str = cbdc::to_string(seckey);
+        std::string pubkey_str = cbdc::to_string(ret);
+        return std::make_pair(seckey_str, pubkey_str);
+    }
+
+    [[nodiscard]] auto config_generator::create_key_pair(const bool randomize)
+        -> std::pair<std::string, std::string> {
+        if(randomize) {
+            return create_random_key_pair();
+        } else {
+            return create_repeatable_key_pair();
+        }
+    }
+
+    [[nodiscard]] auto config_generator::parse_value(const std::string& value,
+                                                     const bool keep_quotes)
+        -> value_t {
+        if(value[0] != '\"' && value[value.size() - 1] != '\"') {
+            if(value.find('.') == std::string::npos) {
+                const auto as_int = static_cast<size_t>(std::stoull(value));
+                return as_int;
+            }
+            const auto as_dbl = std::stod(value);
+            return as_dbl;
+        }
+
+        if(keep_quotes == false) {
+            const auto unquoted = value.substr(1, value.size() - 2);
+            return unquoted;
+        }
+        return value;
+    }
+
+    [[nodiscard]] auto config_generator::get_param_from_template_file(
+        const std::string option,
+        std::map<std::string, std::string>& config_map)
+        -> std::variant<size_t, double, std::string> {
+        auto it = config_map.find(option);
+        if(it != config_map.end()) {
+            value_t parsed_val = parse_value(it->second, false);
+            if(std::holds_alternative<size_t>(parsed_val)) {
+                return std::get<size_t>(parsed_val);
+            } else if(std::holds_alternative<double>(parsed_val)) {
+                return std::get<double>(parsed_val);
+            } else if(std::holds_alternative<std::string>(parsed_val)) {
+                return std::get<std::string>(parsed_val);
+            } else {
+                std::string error_msg
+                    = "Warning: Unrecognized type for param, " + option + ".";
+                std::cout << error_msg << std::endl;
+                return error_msg;
```

Comment:
I simply removed this else statement

Reply:
Used:

 else {
                __builtin_unreachable();
            }

here. Is that correct usage for it? Or do i also need to return something?

NEEDS FOLLOW UP

Reply:
No need to return anything. That intrinsic serves two basic purposes:

* Tells the compiler to avoid diagnostic messages that it might otherwise offer
* Enables the compiler to produce higher-quality assembly by having an assurance that it can elide code paths that would get to that statement.